PatreonClient/Models/IRelationshipExtensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using PatreonClient.Responses;
+
+namespace PatreonClient.Models
+{
+	public static class IRelationshipExtensions
+	{
+		public static IRelationship AssignData<BaseType>(this IRelationship relationship, IReadOnlyCollection<PatreonData> includes, PatreonResponse<BaseType> baseType)
+		{
+			if (baseType?.Data != null)
+			{
+				baseType.Data = includes.FirstOrDefault(x => x.Id == baseType.Data.Id) as PatreonData<BaseType>;
+			}
+
+			return relationship;
+		}
+
+		public static IRelationship AssignDataAndRelationship<BaseType, RelationshipType>(this IRelationship relationship, IReadOnlyCollection<PatreonData> includes, PatreonResponse<BaseType, RelationshipType> baseType) where RelationshipType : IRelationship
+		{
+			if (baseType?.Data != null)
+			{
+				baseType.Data = includes.FirstOrDefault(x => x.Id == baseType.Data.Id) as PatreonData<BaseType, RelationshipType>;
+
+				baseType.Data?.Relationships?.AssignRelationship(includes);
+			}
+
+			return relationship;
+		}
+
+		public static IRelationship AssignCollectionAttributesAndRelationships<BaseType, RelationshipType>(this IRelationship relationship, IReadOnlyCollection<PatreonData> includes, PatreonCollectionResponse<BaseType, RelationshipType> baseCollection) where RelationshipType : IRelationship
+		{
+			if (baseCollection?.Data != null)
+			{
+				foreach (var collectionInstance in baseCollection?.Data)
+				{
+					var include = includes.FirstOrDefault(x => x.Id == collectionInstance.Id) as PatreonData<BaseType, RelationshipType>;
+					if (include != null)
+					{
+						collectionInstance.Attributes = include.Attributes;
+						collectionInstance.Relationships = include.Relationships;
+					}
+
+					collectionInstance.Relationships?.AssignRelationship(includes);
+				}
+			}
+
+			return relationship;
```

Comment:
Might want to think about reversing the if statement and remove redundant null checks...

`
if (baseCollection?.Data == null) return relationship;

foreach (var collectionInstance in baseCollection.Data)
{
   /// ... some code
}
`

Reply:
Good point 1n5an1ty, I've pushed changes to this effect now in commit 3ed0848